src/routes/github/webhook/webhook-receiver-post.ts

```diff
@@ -138,10 +153,31 @@ const getWebhookSecret = async (uuid?: string): Promise<{ webhookSecret: string,
 			throw new Error(`Installation not found for gitHubApp with uuid ${uuid}`);
 		}
 		const webhookSecret = await gitHubServerApp.getDecryptedWebhookSecret(installation.jiraHost);
-		return { webhookSecret, gitHubServerApp };
+		/**
+		 * If we ever need to rotate the webhook secrets for Enterprise Customers,
+		 * we can add it in the array: ` [ webhookSecret ]`
+		 */
+		return { webhookSecrets: [ webhookSecret ], gitHubServerApp };
 	}
-	if (!envVars.WEBHOOK_SECRET) {
-		throw new Error("Environment variable 'WEBHOOK_SECRET' not defined");
+	if (allowGhCloudWebhookSecrets) {
+		if (!envVars.WEBHOOK_SECRETS) {
+			throw new Error("Environment variable 'WEBHOOK_SECRETS' not defined");
+		}
+		try {
+			/**
+			 * The environment WEBHOOK_SECRETS should be a JSON array string in the format: ["key1", "key1"]
+			 * Basically an array of the new as well as any old webhook secrets
+			 */
+			const parsedWebhookSecrets = JSON.parse(envVars.WEBHOOK_SECRETS);
+			return { webhookSecrets: parsedWebhookSecrets };
+		} catch (e) {
+			throw new Error("Couldn't parse 'WEBHOOK_SECRETS', it is not properly defined!");
```

Comment:
I think it is better to parse that in the env.ts, fail-fast, make app crash if that JSON is invalid.
Otherwise, if it fail here, we lose webhook msg.

Reply:
Updated `env.ts` and removed from here